app/src/main/java/io/homeassistant/companion/android/widgets/button/ButtonWidget.kt

```diff
@@ -141,25 +145,36 @@ class ButtonWidget : AppWidgetProvider() {
 
         super.onReceive(context, intent)
         when (action) {
+            CALL_SERVICE_AUTH -> authThenCallConfiguredService(context, appWidgetId)
             CALL_SERVICE -> callConfiguredService(context, appWidgetId)
             RECEIVE_DATA -> saveServiceCallConfiguration(context, intent.extras, appWidgetId)
             Intent.ACTION_SCREEN_ON -> updateAllWidgets(context)
         }
     }
 
+    private fun authThenCallConfiguredService(context: Context, appWidgetId: Int) {
+        Log.d(TAG, "Calling authentication, then configured service")
+
+        val intent = Intent(context, WidgetAuthenticationActivity::class.java)
+        intent.setFlags(Intent.FLAG_ACTIVITY_NEW_TASK)
```

Comment:
If the Home Assistant app was already open in the background, this seems to 'overwrite' the back stack which isn't a great experience. Suggestion: use `Intent.FLAG_ACTIVITY_NEW_TASK or Intent.FLAG_ACTIVITY_NEW_DOCUMENT` to make it use another task + automatically remove it from recents when finished.

Android Studio also suggests using `intent.flags = ...` instead of `intent.setFlags(...)`.

Reply:
OK